### PR TITLE
[1/10] Allow API calls to specify the reply address option

### PR DESCRIPTION
**Pull request by richardc0**
- Added notification_to_email_reply_to link table to the model to link notifications and service_email_reply_to so that notifications are associated to a reply to address.
- Created database migration script to apply the changes.

https://www.pivotaltracker.com/story/show/150440099
Task 1/10

### CODE REVIEW

**Comment by quis** [comment]
You don’t need to change anything on this pull request, but I’d suggest reading https://github.com/alphagov/styleguides/blob/master/git.md

File: app/models.py [hunk 1]
```diff
@@ -1414,3 +1414,24 @@ def serialize(self):
             'created_at': self.created_at.strftime(DATETIME_FORMAT),
             'updated_at': self.updated_at.strftime(DATETIME_FORMAT) if self.updated_at else None
         }
+
+
+class NotificationSmsSender(db.Model):
```
Comment (kentsanggds) [inline app/models.py]: Should probably be NotificationEmailReplyTo?

File: app/models.py [hunk 2]
```diff
@@ -1414,3 +1414,24 @@ def serialize(self):
...
+    __tablename__ = "notification_to_email_sender"
```
Comment (kentsanggds) [inline app/models.py]: Change this to `notification_to_email_reply_to`? Or something like that

File: migrations/versions/0123_add_noti_to_email_sender.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+"""
+
+Revision ID: 0123_add_noti_to_email_sender
```
Comment (kentsanggds) [inline migrations/versions/0123_add_noti_to_email_sender.py]: Update the revision ID

File: migrations/versions/0123_add_noti_to_email_sender.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+Revises: 0122_add_service_letter_contact
+Create Date: 2017-09-27 09:42:39.412731
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+revision = '0123_add_noti_to_email_sender'
```
Comment (kentsanggds) [inline migrations/versions/0123_add_noti_to_email_sender.py]: Update the revision ID

File: app/models.py (hunk 1 above)
Comment (richardc0) [reply app/models.py]: Yes updated

File: app/models.py (hunk 2 above)
Comment (richardc0) [reply app/models.py]: Ok updated

File: migrations/versions/0123_add_noti_to_email_reply.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+def upgrade():
+    op.create_table('notification_to_email_reply_to',
+    sa.Column('notification_id', postgresql.UUID(as_uuid=True), nullable=False),
+    sa.Column('service_email_reply_to_id', postgresql.UUID(as_uuid=True), nullable=False),
+    sa.ForeignKeyConstraint(['notification_id'], ['notifications.id'], ),
+    sa.ForeignKeyConstraint(['service_email_reply_to_id'], ['service_email_reply_to.id'], ),
+    sa.PrimaryKeyConstraint('notification_id', 'service_email_reply_to_id')
+    )
+    op.create_index(op.f('ix_notification_to_email_sender_notification_id'), 'notification_to_email_reply_to', ['notification_id'], unique=False)
```
Comment (servingUpAces) [inline migrations/versions/0123_add_noti_to_email_reply.py]: We expect there to be only one notification_id in the mapping table. We could make the index unique.

**Pull request merged**